### PR TITLE
Use GitHub Actions instead of Travis-CI

### PR DESCRIPTION
**Pull request by janno42**
This moves testing from Travis CI to GitHub Actions.

### CODE REVIEW

File: evap/settings.py [hunk 1]
```diff
@@ -387,7 +387,7 @@
 # and don't want the changes to appear in 'git status'.
 try:
     # if a localsettings file exists (vagrant), this will cause wildcard-import errors
-    # if it does not, (travis), if would cause useless-suppression
+    # if it does not, (GitHub), it would cause useless-suppression
```
Comment (janno42) [inline evap/settings.py:390]: we create a localsettings.py for the tests, right? is this obsolete?

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      matrix:
+        include:
+          - name: "Coverage"
+            command: "coverage run manage.py test && codecov -X gcov"
+          - name: "Debug mode"
+            command: "python manage.py test --debug-mode"
+          - name: "Reverse order"
+            command: "python manage.py test --reverse"
+          #- name: "Backup process"
+          #  command: "deployment/test_backup_and_restore_on_github.sh"
```
Comment (janno42) [inline .github/workflows/tests.yml:37]: this is not yet working, will look into it

Comment (janno42) [reply .github/workflows/tests.yml:37]: @He3lixxx, it looks like loading the backup starts already before the update script is done. could you have a look at it? ( https://github.com/janno42/EvaP/runs/1814767358?check_suite_focus=true)

File: deployment/load_production_backup.sh [hunk 3]
```diff
@@ -9,7 +9,7 @@ cd $(dirname $0)/.. # change to root directory
 USERNAME="evap"
 ENVDIR="/opt/evap/env"
 ADDITIONAL_ARGUMENTS=""
-[[ ! -z "$EVAP_RUNNING_INSIDE_TRAVIS" ]] && echo "Detected travis" && USERNAME="travis" && ENVDIR=~/virtualenv/python3.7 && ADDITIONAL_ARGUMENTS=" --noinput"
+[[ ! -z "$GITHUB_WORKFLOW" ]] && echo "Detected GitHub" && USERNAME="github" && ENVDIR=/usr/local && ADDITIONAL_ARGUMENTS=" --noinput"
```
Comment (janno42) [inline deployment/load_production_backup.sh:12]: i don't know which `USERNAME` to put here and in `update_production.sh`

Comment (janno42) [reply deployment/load_production_backup.sh:12]: when not specifying the user (by removing all the `sudo -H -u $USERNAME` parts) everything runs nicely, see https://github.com/janno42/EvaP/runs/1816468720

**Review by He3lixxx — APPROVED** [review]
Thank you :)

**Comment by karyon** [comment]
Sorry for not reviewing earlier! I just wanted to say this is really cool, nice job :)

**Pull request merged**